src/web.ts

```diff
@@ -1,22 +1,115 @@
-import { WebPlugin } from '@capacitor/core';
-import { SignInWithApplePlugin } from './definitions';
+import { WebPlugin } from "@capacitor/core";
+
+import {
+  SignInWithApplePlugin,
+  InitOptions,
+  SignInResponse,
+  SignInError,
+} from "./definitions";
+
+declare var window: any;
+
+export class SignInWithAppleWeb extends WebPlugin
+  implements SignInWithApplePlugin {
+  public ready: Promise<any>;
+  private readyResolver: Function;
+  private hasInitialised = false;
+
+  private APPLE_SCRIPT_KEY = "apple_signin_script";
 
-export class SignInWithAppleWeb extends WebPlugin implements SignInWithApplePlugin {
   constructor() {
     super({
-      name: 'SignInWithApple',
-      platforms: ['web']
+      name: "SignInWithApple",
+      platforms: ["web"],
     });
+
+    this.ready = new Promise((resolve) => (this.readyResolver = resolve));
+    this.init();
+  }
+
+  async init() {
```

Comment:
Ok, this naming was an example, rename it so it doesn't collide with the public API `Init` method below